### PR TITLE
Cleaning up previous runs from GPU runner

### PR DESCRIPTION
**Pull request by agunapal**
## Description

This PR cleans up the workspace directory fully from the CI GPU runners.

Fixes #(issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Feature/Issue validation/testing

Please describe the Unit or Integration tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
Logs for Test A

- [ ] Test B
Logs for Test B

## Checklist:

- [ ] Did you have fun?
- [ ] Have you added tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

### CODE REVIEW

**Comment by msaroufim** [comment]
Let's wait on ci, might also need to skip the streaming handler tests to actually merge this

I would have expected `cd $RUNNER_WORKSPACE` to stick around

File: .github/workflows/ci_gpu.yml [hunk 1]
```diff
@@ -20,11 +20,10 @@ jobs:
...
           echo "Cleaning up previous run"
-          cd $RUNNER_WORKSPACE
-          pwd
-          cd ..
-          pwd
-          rm -rf _tool
+          ls -la ./
+          sudo rm -rf ./* || true
+          sudo rm -rf ./.??* || true
+          ls -la ./
```
Comment (lxning) [inline .github/workflows/ci_gpu.yml:26]: we need specify the path and then delete

Comment (agunapal) [reply .github/workflows/ci_gpu.yml:26]: This doesn't work https://github.com/pytorch/serve/actions/runs/6409297939

Also, we have been doing the same in Regression GPU runners and it works     

https://github.com/pytorch/serve/blob/master/.github/workflows/regression_tests_gpu.yml#L21

Comment (msaroufim) [reply .github/workflows/ci_gpu.yml:26]: Could you maybe add a comment to more explicitly say what this is trying to achieve?

The benefit of not having `*` in remove commands is we can be more explicit about what's being deleted

**Comment by msaroufim** [comment]
Just so we can move on 

```bash
# deletes all files in the current directory
sudo rm -rf ./* || true
# deletes all hidden files in a directory
sudo rm -rf ./.??* || true
``

**Review by msaroufim — APPROVED** [review]

**Pull request merged**